src/compas_timber/connections/half_lap.py

```diff
@@ -0,0 +1,152 @@
+from compas_timber.connections import Joint
+from compas_timber.connections import JointTopology
+from compas.geometry import Point, Line, Plane, Polyhedron, Brep
+from compas.geometry import midpoint_point_point
+from compas.geometry import intersection_plane_plane
+from compas.geometry import intersection_line_plane
+from compas.geometry import angle_vectors
+from compas.artists import Artist
+from compas_timber.parts import BeamBooleanSubtraction
+from compas_timber.utils import intersection_line_line_3D
+
+# BREP HACK CHEN
+import rhinoscriptsyntax as rs
+from Rhino.Geometry import Brep as RhinoBrep
+
+
+class HalfLapJoint(Joint):
+    SUPPORTED_TOPOLOGY = JointTopology.TOPO_X
+
+    def __init__(self, assembly=None, beam_a=None, beam_b=None, cut_plane_choice=None):
+        super(HalfLapJoint, self).__init__(assembly, [beam_a, beam_b])
+        self.beam_a = beam_a
+        self.beam_b = beam_b
+        self.beam_a_key = None
+        self.beam_b_key = None
+        self.cut_plane_choice = cut_plane_choice  # Decide if Direction of beam_a or beam_b
+        self.max_distance = (beam_a.height + beam_b.height) * 0.55  # A Bit more than the Average of both Thicknesses
+        self.features = []
+
+    @property
+    def data(self):
+        data_dict = {
+            "beam_a": self.beam_a.key,
+            "beam_b": self.beam_b.key,
+        }
+        data_dict.update(Joint.data.fget(self))
+        return data_dict
+
+    @data.setter
+    def data(self, value):
+        Joint.data.fset(self, value)
+        self.beam_a_key = value["beam_a"]
+        self.beam_b_key = value["beam_b"]
+
+    @property
+    def joint_type(self):
+        return "Half-Lap"
+
+    @property
+    def beams(self):
+        return [self.beam_a, self.beam_b]
+
+    def _cutplane(self):
+        # Sets the Plane where the Cut will be
+        centerline_a = self.beam_a.centerline
+        centerline_b = self.beam_b.centerline
+        plane_a = Plane.from_frame(self.beam_a.faces[3])
+        plane_b = Plane.from_frame(self.beam_b.faces[3])
+        int_a, int_b = intersection_line_line_3D(centerline_a, centerline_b, self.max_distance)
+        int_a, _ = int_a
+        int_b, _ = int_b
+        point_cut = Point(*midpoint_point_point(int_a, int_b))  # If Centerline don't intersect
+        if self.cut_plane_choice == None or "A":
+            plane = plane_a
+        if self.cut_plane_choice == "B":
+            plane = plane_b
+        plane_cut = Plane(point_cut, plane[1])
+        return plane_cut
+
+    @staticmethod
+    def _create_polyhedron(plane_top, plane_bottom, lines_vertical):
```

Comment:
why "_vertical" ? 
I'd strongly recommend to make this method generic so that you can add other orientations of splitting planes.